climpred/tests/test_bootstrap.py

```diff
@@ -0,0 +1,141 @@
+import time
+
+import dask
+import pytest
+from xarray.testing import assert_allclose
+
+from climpred.bootstrap import bootstrap_hindcast, bootstrap_perfect_model, my_quantile
+from climpred.constants import HINDCAST_COMPARISONS, PM_COMPARISONS
+from climpred.tutorial import load_dataset
+
+
+@pytest.fixture
+def pm_da_ds1d():
+    da = load_dataset('MPI-PM-DP-1D')
+    da = da['tos'].isel(area=1, period=-1)
+    return da
+
+
+@pytest.fixture
+def pm_da_control1d():
+    da = load_dataset('MPI-control-1D')
+    da = da['tos'].isel(area=1, period=-1)
+    return da
+
+
+@pytest.fixture
+def initialized_da():
+    da = load_dataset('CESM-DP-SST')['SST']
+    da = da - da.mean('init')
+    return da
+
+
+@pytest.fixture
+def uninitialized_da():
+    da = load_dataset('CESM-LE')['SST']
+    # add member coordinate
+    da['member'] = range(1, 1 + da.member.size)
+    da = da - da.mean('time')
+    return da
+
+
+@pytest.fixture
+def observations_da():
+    da = load_dataset('ERSST')['SST']
+    da = da - da.mean('time')
+    return da
+
+
+@pytest.fixture
+def ds3d():
+    """ds3d"""
+    ds3d = load_dataset('MPI-PM-DP-3D')['tos']
+    return ds3d
+
+
+@pytest.fixture
+def control3d():
+    """control 3d"""
+    control3d = load_dataset('MPI-control-3D')['tos']
+    return control3d
+
+
+# @pytest.mark.parametrize('dims', (('x', 'time'), ('time', 'x')))
+@pytest.mark.parametrize('chunk', [True, False])
+def test_dask_percentile_implemented_faster_xr_quantile(control3d, chunk):
+    chunk_dim, dim = 'x', 'time'
+    # chunk_dim, dim = 'time', 'x' # fails, why?
```

Comment:
`my_quantile` works as expected in `bootstrap_compute` because it is applied to the first dimension (there `bootstrap`, here `time`). but it fails to work on axis!=0 with nans.

this small demo runs without asserion errors:
```python
a.dims
a.shape

q = .05

chunk_dim, dim = 'lon', 'time'
ac = a.chunk({chunk_dim: 2})
acp = a.quantile(dim=dim,q=q)
acp.shape
%time acp2 = my_quantile(ac, dim, q).compute()
acp2.shape
xr.testing.assert_allclose(acp, acp2)
acl = ac.load()
%time acp = my_quantile(acl, dim, q).compute()
acp.shape
xr.testing.assert_allclose(acp, acp2)


chunk_dim, dim = 'time', 'lon'
acp = a.quantile(dim=dim,q=q)
acp.shape
ac = a.chunk({chunk_dim: 2})
%time acp2 = my_quantile(ac, dim, q).compute()
acp2.shape
xr.testing.assert_allclose(acp, acp2)
acl = ac.load()
%time acp = my_quantile(acl, dim, q).compute()
acp.shape
xr.testing.assert_allclose(acp, acp2)
```